lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_clear_intermediate_tags.rb

```diff
@@ -5,18 +5,20 @@ def self.run(params)
         UI.message("Deleting tags for version: #{params[:version]}")
 
         require_relative '../../helper/git_helper.rb'
-        Fastlane::Helper::Ios::GitHelper.delete_tags(params[:version])
 
-        # Cleanup local tags and refetch them to make sure we're up-to-date with remote
-        local_tags = Fastlane::Helper::GitHelper.list_local_tags()
-        Fastlane::Helper::GitHelper.delete_tags(local_tags, delete_on_remote: true)
-        Fastlane::Helper::GitHelper.fetch_all_tags()
+        # Delete the current version's tag (if it exists)
+        Fastlane::Helper::GitHelper.delete_tags(params[:version])
 
-        # Now delete intermediate tags (4-parts version names starting with our version number)
+        # Delete 4-parts version names starting with our version number
         parts = params[:version].split('.')
         pattern = parts.fill('*', parts.length...4).join('.') # "1.2.*.*" or "1.2.3.*"
+
         intermediate_tags = Fastlane::Helper::GitHelper.list_local_tags(matching: pattern)
-        Fastlane::Helper::GitHelper.delete_tags(intermediate_tags, delete_on_remote: true)
+        tag_count = intermediate_tags.count
+
+        if tag_count > 0 && UI.confirm("Are you sure you want to delete #{tag_count} tags?")
+          Fastlane::Helper::GitHelper.delete_tags(intermediate_tags, delete_on_remote: true)
+        end
```

Comment:
What about listing the actual list of tags that we are gonna delete instead?
```suggestion
        unless intermediate_tags.empty?
          UI.message("About to delete the following intermediate tags:" + intermediate_tags.map { |t| " - #{t}" }.join("\n")
          if UI.confirm("Are you sure you want to delete those tags?")
            Fastlane::Helper::GitHelper.delete_tags(intermediate_tags, delete_on_remote: true)
          end
        end
```

Reply:
This appears in the `fastlane` output for me like this:

```
[12:51:29]: -----------------------------------------
[12:51:29]: --- Step: ios_clear_intermediate_tags ---
[12:51:29]: -----------------------------------------
[12:51:29]: To not be asked about this value, you can specify it using 'version'
[12:51:29]: The version of the tags to clear: 16.9
[12:51:31]: Deleting tags for version: 16.9
16.9
[12:51:32]: $ git tag --list 16.9.\*.\*
[12:51:32]: ▸ 16.9.0.0
[12:51:32]: ▸ 16.9.0.1
```

Reply:
Ah, nice.
🤔  That logging will likely disappear if we switch from `Action.sh` to `Git` gem to also list the tags in the future though, right?

Reply:
Yeah that's correct – right now it's sort of a side-effect that we're relying on. For now it's IMHO ok, but at some point we should probably run around clean it up (though I don't think this is necessarily the time to do so)